Core/Layer/Worlds/StatusBar/StatusBarRenderer.cs

```diff
@@ -0,0 +1,1160 @@
+    {
+        if (!StatusBarConditionResolver.Evaluate(context, faceBg.Conditions))
+            return;
+
+        Vec2I currentPos = ResolvePosition(faceBg, parentPos, widescreenOffset);
+        string patch = GetFaceBackgroundPatch(context.Player);
+
+        if (!string.IsNullOrEmpty(patch))
+        {
+            Align align = ConvertAlignment(faceBg.Alignment);
+            float alpha = faceBg.Translucency ? 0.5f : 1.0f;
+            DrawSBarTexture(hud, patch, currentPos, align, useDoomOffsets: true, 
+                translation: faceBg.Translation, alpha: alpha);
+        }
+
+        DrawChildren(hud, faceBg, currentPos, containerHeight, context, widescreenOffset);
+    }
+
+    private void DrawAnimation(IHudRenderContext hud, StatusBarAnimationDef anim, Vec2I parentPos, int containerHeight, StatusBarContext context, float widescreenOffset)
+    {
+        if (!StatusBarConditionResolver.Evaluate(context, anim.Conditions))
+            return;
+
+        Vec2I currentPos = ResolvePosition(anim, parentPos, widescreenOffset);
+
+        if (anim.Frames.Count > 0)
+        {
+            double totalDuration = anim.Frames.Sum(f => f.Duration);
+            if (totalDuration > 0)
+            {
+                double timePerLoop = totalDuration * Constants.TicksPerSecond;
+                long currentTick = WorldStatic.World.LevelTime;
+                double animTime = currentTick % timePerLoop;
+
+                string patch = anim.Frames[0].Lump;
+                double timeAccumulator = 0;
+
+                foreach (var frame in anim.Frames)
+                {
+                    timeAccumulator += frame.Duration * Constants.TicksPerSecond;
+                    if (animTime < timeAccumulator)
+                    {
+                        patch = frame.Lump;
+                        break;
+                    }
+                }
+
+                Align align = ConvertAlignment(anim.Alignment);
+                DrawSBarTexture(hud, patch, currentPos, align, useDoomOffsets: true, 
+                    translation: anim.Translation, alpha: 1.0f);
+            }
+        }
+
+        DrawChildren(hud, anim, currentPos, containerHeight, context, widescreenOffset);
+    }
+
+    private void DrawComponent(IHudRenderContext hud, StatusBarComponentDef comp, Vec2I parentPos, int containerHeight, StatusBarContext context, float widescreenOffset)
+    {
+        if (!StatusBarConditionResolver.Evaluate(context, comp.Conditions))
+            return;
+
+        Vec2I pos = ResolvePosition(comp, parentPos, widescreenOffset);
+        string text = string.Empty;
+        var config = WorldStatic.World.Config.Hud;
+        
+        float alpha = comp.Translucency ? 0.5f : 1.0f;
+        StatusBarAlignment alignment = comp.Alignment;
+
+        var sbarDef = WorldStatic.World.ArchiveCollection.Definitions.StatusBarDefinition;
+        var fontDef = sbarDef.HudFonts.FirstOrDefault(f => f.Name.EqualsIgnoreCase(comp.Font));
```

Comment:
Since this is in the render loop, to be safe let's extract this out to a function with a manual for loop iteration. This is just being extra safe since LINQ functions generally allocate memory that would leak to the garbage collector.